lib/Doctrine/Migrations/DependencyFactory.php

```diff
@@ -136,6 +136,15 @@ public function hasEntityManager() : bool
         return $this->emLoader !== null;
     }
 
+    public function setConfigurationLoader(ConfigurationLoader $configurationLoader) : void
+    {
+        if ($this->configuration !== null) {
+            throw FrozenDependencies::new();
```

Comment:
Here a better error message will be provided

Reply:
Should there be also a check if the configuration is frozen? Being `!== null` doesn't mean it's already frozen.